herd/ASLSem.ml

```diff
@@ -507,7 +509,8 @@ module Make (C : Config) = struct
         let bind_data = M.( >>= )
         let bind_seq = M.aslseq
         let bind_ctrl = M.bind_ctrl_seq_data
-        let prod = M.( >>| )
+        let bind_par = M.( >>| )
+        let appl m f = m >>= fun v -> return (f v)
```

Comment:
@maranget this is out of scope for this PR but maybe it would be interesting to implement this into `herd/monad.mli`?